### PR TITLE
Slim down docker image for miniconda

### PR DESCRIPTION
**Pull request by soapy1**
Going from ~574MB to ~390MB

### CODE REVIEW

File: miniconda/debian/Dockerfile [hunk 1]
```diff
@@ -1,32 +1,20 @@
...
+FROM debian:buster-slim
 
 ENV LANG=C.UTF-8 LC_ALL=C.UTF-8
-ENV PATH /opt/conda/bin:$PATH
 
-RUN apt-get update --fix-missing && apt-get install -y wget bzip2 ca-certificates \
-    libglib2.0-0 libxext6 libsm6 libxrender1 \
-    git mercurial subversion
+RUN apt-get update --fix-missing && \
+    apt-get install -y wget bzip2 ca-certificates libglib2.0-0 libxext6 libsm6 libxrender1 git mercurial subversion && \
```
Comment (jcrist) [inline miniconda/debian/Dockerfile]: Git and friends are not small applications, and aren't often needed in deployments. Considering these can be conda installed, should all these packages be part of a base conda image? FWIW this is the base conda image I've been using for things: https://github.com/jcrist/alpine-dask-docker/blob/master/alpine-conda/Dockerfile. It's just `conda` and `tini` for init, nothing else. All other things can be added by subsequent layers.

If these development tools are needed (perhaps for backwards compat), they could maybe be added at a higher layer, and a new image ("conda-base" or something) that contains only conda could be created.

File: miniconda/debian/Dockerfile [hunk 2]
```diff
@@ -1,32 +1,20 @@
...
-
-RUN apt-get install -y curl grep sed dpkg && \
-    TINI_VERSION=`curl https://github.com/krallin/tini/releases/latest | grep -o "/v.*\"" | sed 's:^..\(.*\).$:\1:'` && \
-    curl -L "https://github.com/krallin/tini/releases/download/v${TINI_VERSION}/tini_${TINI_VERSION}.deb" > tini.deb && \
-    dpkg -i tini.deb && \
-    rm tini.deb && \
-    apt-get clean
+    echo "conda activate base" >> ~/.bashrc && \
+    find /opt/conda/ -follow -type f -name '*.a' -delete && \
+    find /opt/conda/ -follow -type f -name '*.pyc' -delete && \
```
Comment (jcrist) [inline miniconda/debian/Dockerfile]: `*.pyc` files aren't small, but if you don't prevent python from recreating them on startup (e.g. by defining `PYTHONDONTWRITEBYTECODE`) then you pay a runtime cost from the overlay filesystem (which may not be cheap). I'm not 100% convinced removing these is worth it, that blogpost was mainly an experiment to see how small a conda image could be.

File: miniconda/debian/Dockerfile (hunk 1 above)
Comment (soapy1) [reply miniconda/debian/Dockerfile]: A lot of users seem to use these images for CI so I think these tools need to be here for backwards compatibility. I like the idea of having a conda-base image but I also think that having an extra layer of images will add a perhaps unneeded level of complexity to the build/release process for these images. 
Building without git, etc. beings the image down to 265MB. Do you see any other use cases for a "conda-base" debian based image? For people who want a more lean conda docker image without any extra frills there is also a conda alpine image: https://github.com/ContinuumIO/docker-images/blob/master/miniconda/alpine/Dockerfile

Comment (jcrist) [reply miniconda/debian/Dockerfile]: The benefit of having a standard conda base image that users can depend on is that the layers in the conda base image are shared, minimizing total disk space. The downside of maintaining such an image is you'd need to appease everyone, which is always tricky. If the main users of the conda image are CI/development cases, then what you have now seems fine.

**Pull request merged**